types.go

```diff
@@ -23,15 +23,34 @@ import (
 )
 
 type Request struct {
-	Service string `protobuf:"bytes,1,opt,name=service,proto3"`
-	Method  string `protobuf:"bytes,2,opt,name=method,proto3"`
-	Payload []byte `protobuf:"bytes,3,opt,name=payload,proto3"`
+	Service  string             `protobuf:"bytes,1,opt,name=service,proto3"`
+	Method   string             `protobuf:"bytes,2,opt,name=method,proto3"`
+	Payload  []byte             `protobuf:"bytes,3,opt,name=payload,proto3"`
+	Metadata map[string]*Values `protobuf:"bytes,4,rep,name=metadata,proto3" protobuf_key:"bytes,1,opt,name=key,proto3" protobuf_val:"bytes,2,opt,name=value,proto3"`
```

Comment:
Just make this compatible with the auto generated map type. Should be a string key with a bytes value.

Reply:
Will double check this, I used protoc to generate, but may have messed something up along the way.

Reply:
What'd you generate from? Looking at https://github.com/containerd/containerd/blob/master/api/services/content/v1/content.proto#L93, it should just generate a map to Value.

Since we really want an ordered multimap, would just having a repeated Values as a slice work here?

Reply:
> What'd you generate from?

I just created a proto for `Request` with my added stuff.

```
type Request struct {
	Service              string             `protobuf:"bytes,1,opt,name=service,proto3" json:"service,omitempty"`
	Method               string             `protobuf:"bytes,2,opt,name=method,proto3" json:"method,omitempty"`
	Payload              []byte             `protobuf:"bytes,3,opt,name=payload,proto3" json:"payload,omitempty"`
	Metadata             map[string]*Values `protobuf:"bytes,4,rep,name=metadata,proto3" json:"metadata,omitempty" protobuf_key:"bytes,1,opt,name=key,proto3" protobuf_val:"bytes,2,opt,name=value,proto3"`
	XXX_NoUnkeyedLiteral struct{}           `json:"-"`
	XXX_unrecognized     []byte             `json:"-"`
	XXX_sizecache        int32              `json:"-"`
}
```

> Since we really want an ordered multimap

WDYM?

> would just having a repeated Values as a slice work here?

I think we want an index here?

Reply:
Ok, this looks fine then.